src/main/scala/com/databricks/labs/overwatch/MultiWorkspaceDeployment.scala

```diff
@@ -0,0 +1,369 @@
+package com.databricks.labs.overwatch
+
+import com.databricks.labs.overwatch.pipeline._
+import com.databricks.labs.overwatch.utils._
+import com.databricks.labs.overwatch.validation.DeploymentValidation
+import org.apache.log4j.{Level, Logger}
+import org.apache.spark.sql.Dataset
+import org.apache.spark.sql.functions.lit
+
+import java.text.SimpleDateFormat
+import java.time.LocalDateTime
+import java.util
+import java.util.{Collections, Date}
+import java.util.concurrent.Executors
+import scala.collection.mutable.ArrayBuffer
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutor, Future}
+
+/** *
+ * MultiWorkspaceDeployment class is the main class which runs the deployment for multiple workspaces.
+ *
+ * @params configCsvPath: path of the csv file which will contain the different configs for the workspaces.
+ * @params tempOutputPath: location which will be used as a temp storage.It will be automatically cleaned after each run.
+ * @params apiEnvConfig: configs related to api call.
+ */
+object MultiWorkspaceDeployment extends SparkSessionWrapper {
+
+
+
+  def apply(configCsvPath: String): MultiWorkspaceDeployment = {
+   apply(configCsvPath,"/mnt/tmp/overwatch")
+  }
+
+  def apply(configCsvPath: String, tempOutputPath: String)={
+    new MultiWorkspaceDeployment()
+      .setConfigCsvPath(configCsvPath)
+      .setOutputPath(tempOutputPath)
+      .setPipelineSnapTime()
+  }
+}
```

Comment:
This is where I think we should build the strongly typed DS and use that throughout instead of continually converting the CSV to the DF/DS/Rows